pkg/stub/replset.go

```diff
@@ -132,6 +133,78 @@ func (h *Handler) handleReplsetInit(m *v1alpha1.PerconaServerMongoDB, replset *v
 	return ErrNoRunningMongodContainers
 }
 
+func (h *Handler) handleStatefulSetUpdate(m *v1alpha1.PerconaServerMongoDB, set *appsv1.StatefulSet, replset *v1alpha1.ReplsetSpec, resources *corev1.ResourceRequirements) error {
+	var doUpdate bool
+
+	// Ensure the stateful set size is the same as the spec
+	err := h.client.Get(set)
+	if err != nil {
+		return fmt.Errorf("failed to get stateful set for replset %s: %v", replset.Name, err)
+	}
+	if *set.Spec.Replicas != replset.Size {
+		logrus.Infof("setting replicas to %d for replset: %s", replset.Size, replset.Name)
+		set.Spec.Replicas = &replset.Size
+		doUpdate = true
+	}
+
+	// Ensure the PSMDB version is the same as the spec
+	expectedImage := getPSMDBDockerImageName(m)
+	mongod := &set.Spec.Template.Spec.Containers[0]
```

Comment:
Accessing to the slice by index without checks seems to me unsafe. It may lead to a runtime crash. Now it shouldn't, because, as I see, the StatefulSet always came to this method with some container. But later on, some logic might be changed and the `index out of range` may happen.